src/app/pages/storage2/modules/devices/components/devices/devices.component.ts

```diff
@@ -104,12 +114,30 @@ export class DevicesComponent implements OnInit {
     this.selectedParentItem = undefined;
   }
 
-  onRowSelected(dataNodeSelected: DeviceNestedDataNode, event: MouseEvent): void {
-    event.stopPropagation();
-    this.selectedItem = dataNodeSelected;
-    this.selectedParentItem = findInTree(this.treeControl.dataNodes, (dataNode: DeviceNestedDataNode) => {
-      return dataNode.guid === dataNodeSelected.guid;
-    });
+  private listenForRouteChanges(id: string): void {
+    this.selectedItem = undefined;
+    this.selectedParentItem = undefined;
+
+    if (id && this.dataSource?.data) {
```

Comment:
Suggestion for a minor improvement - can we apply "return early" principle here? Just to improve readability.